CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Networking/DocumentService.swift

```diff
@@ -27,34 +52,57 @@ public final class DocumentService: DocumentServiceProtocol {
             .map { $0.value }
             .sorted()
             .map { $0.info }
-        
-        self.fetchExtractions(for: partialDocumentsInfoSorted, completion: completion)
+        self.analysisCancellationToken = CancellationToken()
+        captureNetworkService.analyse(partialDocuments: partialDocumentsInfoSorted, metadata: metadata, cancellationToken: analysisCancellationToken!) { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case let .success((createdDocument, extractionResult)):
+                self.document = createdDocument
+                completion(.success(extractionResult))
+            case let .failure(error):
+                if error != .requestCancelled {
+                    completion(.failure(error))
+                }
+            }
+        }
     }
     
     public func cancelAnalysis() {
         if let compositeDocument = document {
-            delete(compositeDocument)
+            captureNetworkService.delete(document: compositeDocument) {[weak self] result in
+                switch result {
+                case .success:
+                    self?.analysisCancellationToken?.cancel()
```

Comment:
Seeing the old code and the new one side-by-side I noticed, that the old code always invoked the following three lines:
```
self?.analysisCancellationToken?.cancel()
self?.analysisCancellationToken = nil
self?.document = nil
```

It did not check whether there was a document or wether deletion succeeded or not. I recommend moving these lines back outside of the callback and after the if block.